### ISSUE TITLE
ResolvedPackageReference item's DiagnosticLevel metadata not populated when targeting .NET 5

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/sdk#13518]
This bug is similar in spirit to that fixed in https://github.com/dotnet/sdk/pull/13093

One responsibility of the `ResolvePackageDependencies` task is to determine the maximum diagnostic level associated with the package reference. It does this by scanning log messages in from the assets file, matching them to package identifiers.

https://github.com/dotnet/sdk/blob/029c729603a1bf14875187f6a461830d4f3b8da0/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs#L254-L266

Part of that matching is a comparison of target framework. In .NET 5, this task compares strings `net5.0` with `.NETCoreApp,Version=v5.0`, which results in filtering out all log messages.

![image](https://user-images.githubusercontent.com/350947/92842868-6f835000-f427-11ea-8d3a-36d37c2699c6.png)

The end result is that a package's diagnostics are not presented on its node in the tree, which makes identifying a problem harder:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="NuGet.Commands" Version="4.0.0" />
    <PackageReference Include="NuGet.Packaging" Version="3.5.0" />
  </ItemGroup>

</Project>
```

Warning icons should be visible where highlighted:

![image](https://user-images.githubusercontent.com/350947/92842686-364ae000-f427-11ea-8d3f-47a697ee97de.png)

**Comment by sfoslund** [comment]
Looks like this was fixed by https://github.com/dotnet/sdk/pull/13598, right @dsplaisted?

**Comment by dsplaisted** [comment]
> Looks like this was fixed by #13598, right @dsplaisted?

Yes, correct.
